### PR TITLE
Add pagination to hogwarts table

### PR DESCRIPTION
**Pull request by reesercollins**
NOTE: AgGrid doesn't re-render on the gridOptions changing, so the query has to be rerun for the changes to take effect. Not much we can do to fix that AFAIK.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://cccs.atlassian.net/browse/CLDN-1428
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by cccs-nik — COMMENTED** [review]
I see you're working on this actively, but I for sure think changing this setting needs to properly refresh the grid. Doesn't seem to be working yet for me locally even with your latest commits.

Seems to work fine for the example in the documentation so I'm sure it's doable 😋 https://www.ag-grid.com/javascript-data-grid/row-pagination/#example-customising-pagination

**Comment by cccs-nik** [comment]
When opening an existing chart using the Hogwarts Table (don't mind my tags): 
image

I assume the existing pagination feature in the vanilla Superset table doesn't use a validator for this reason.

**Comment by cccs-rc** [comment]
Without the validator we get this: 
image
Are you still working on this @reesercollins ?

**Comment by reesercollins** [comment]
> Without the validator we get this: image Are you still working on this @reesercollins ?

It seems like the default table viz does not use a validator for the reason shown above by Nik.

**Comment by cccs-rc** [comment]
> > Without the validator we get this: image Are you still working on this @reesercollins ?
> 
> It seems like the default table viz does not use a validator for the reason shown above by Nik.

But the default table viz also doesn't seem to give me that error. It just stops me from entering non-numeric values. Can we replicate that behaviour?

**Comment by cccs-nik** [comment]
Now with the validator I can't select "All" (0) anymore. Looks like the validator is only for positive non-zero integers. Here's my suggestion, let me know if it works: Set default to 0, use legacyValidateInteger instead, and modify the following in cccs-grid:

```
const gridRef = useRef<AgGridReact>(null);
const keyRefresh = useRef<number>(0);
const pageSize = useRef<number>(page_length);
```
```
useEffect(() => {
  pageSize.current = page_length <= 0 ? 0 : page_length;
  gridRef.current?.props.api?.paginationSetPageSize(pageSize.current);
  keyRefresh.current += 1;
}, [page_length]);
```
```
paginationPageSize={pageSize.current}
pagination={pageSize.current !== 0}
```

Also I think it's weird setting the pagination size doesn't re-render the grid. It should but I tried to force it to and it wouldn't work. Your solution looks good enough though and it works.

**Review by cccs-RyanS — APPROVED** [review]
LGTM

File: superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx [hunk 1]
```diff
@@ -520,8 +531,23 @@ if (isFeatureEnabled(FeatureFlag.DASHBOARD_CROSS_FILTERS)) {
...
+          name: 'page_length',
+          config: {
+            type: 'SelectControl',
+            freeForm: true,
+            renderTrigger: true,
+            label: t('Page length'),
+            default: 100,
+            choices: PAGE_SIZE_OPTIONS,
+            description: t('Rows per page, 0 means no pagination'),
+            validators: [legacyValidateInteger]
```
Comment (cccs-nik) [inline superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx:547]:
```suggestion
            validators: [legacyValidateInteger],
```

File: superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx [hunk 2]
```diff
@@ -36,11 +37,21 @@ import {
   sharedControls,
   ControlPanelState,
   ControlState,
+  formatSelectOptions
```
Comment (cccs-nik) [inline superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx:40]:
```suggestion
  formatSelectOptions,
```

**Review by cccs-nik — APPROVED** [review]

**Review by cccs-rc — APPROVED** [review]

**Pull request merged**